core/domain/summary_services.py

```diff
@@ -163,6 +163,72 @@ def get_displayable_collection_summary_dicts_matching_ids(collection_ids):
     return _get_displayable_collection_summary_dicts(collection_summaries)
 
 
+def get_collection_node_metadata_dicts(query_string, search_cursor):
+    """Given a query string and a search cursor, returns a list of exploration
+    dicts that satisfy the search query.
+
+    Args:
+        query_string: str. The search query for which the search is to be
+            performed.
+        search_cursor: The cursor location for next search.
```

Comment:
str or None. The cursor location to start the search from. If None, the returned values are from the beginning of the results list.